pkg/performanceprofile/profilecreator/profilecreator_test.go

```diff
@@ -1483,6 +1483,63 @@ var _ = Describe("PerformanceProfileCreator: Test Helper Function ensureSameTopo
 			err := ensureSameTopology(&originTopology, &mutatedTopology)
 			Expect(err).ToNot(HaveOccurred())
 		})
+		It("should fail when core id on same numa is associated with different siblings list", func() {
```

Comment:
can we please add a comment documenting on which kind of hardware we found this happening?
e.g. server Dell model So-and-So with N Intel CPUs model So-and-So.

Reply:
Thanks for the comment. This was never observed and likely never happens on a healthy system but still we want to have this case covered because we depend on must-gather to get us the data for the PPC to analyze as stated in the commit message. This check can be considered as a backup for tolerating the core ID difference, so we reference to core ID in the NUMA level and verify alignment to same CPU siblings each time especially when HT is enabled.